### PR TITLE
kube-aws: validate that the key pair exists

### LINKED ISSUE #331: Validate should check if the key pair exists.
Without a valid key pair `kube-aws up` will fail after several minutes. 

```Error creating cluster: Stack creation failed: CREATE_FAILED :  The following resource(s) failed to create: [LaunchConfigurationWorker, InstanceController].```

We should be able to check for that quickly check for the key pair before attempting to create the cluster.

### PR DESCRIPTION
**Pull request by cgag**
fixes #331

### CODE REVIEW

**Comment by colhom** [comment]
#346 exposes a common aws session via the cluster.Cluster struct. Will wait on that.

**Comment by cgag** [comment]
Nice, thought that session stuff smelled a little funny.

**Comment by cgag** [comment]
Had to move some stuff around to avoid a cyclic dependency.  Now that session is in config, the cluster.Cluster struct seems pretty pointless but I left it around to minimize the scope of this PR.

**Comment by colhom** [comment]
@cgag drop the last commit and we should be good to go

File: multi-node/aws/pkg/config/config.go [hunk 1]
```diff
@@ -193,6 +195,27 @@ func (c Cluster) stackConfig(opts StackTemplateOptions, compressUserData bool) (
 	return &stackConfig, nil
 }
 
+func (c Cluster) ValidateKey() error {
```
Comment (colhom) [inline multi-node/aws/pkg/config/config.go]: Move this to cluster package.

File: multi-node/aws/cmd/kube-aws/command_validate.go [hunk 2]
```diff
@@ -34,6 +34,13 @@ func runCmdValidate(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("Unable to load cluster config: %v", err)
 	}
 
+	fmt.Printf("Validating Key Pair...\n")
```
Comment (colhom) [inline multi-node/aws/cmd/kube-aws/command_validate.go]: Move this logic to https://github.com/coreos/coreos-kubernetes/blob/master/multi-node/aws/pkg/cluster/cluster.go#L56

File: multi-node/aws/cmd/kube-aws/command_validate.go [hunk 3]
```diff
@@ -54,8 +53,10 @@ func runCmdValidate(cmd *cobra.Command, args []string) error {
 	}
 
 	if err != nil {
-		return fmt.Errorf("Error creating cluster: %v", err)
```
Comment (cgag) [inline multi-node/aws/cmd/kube-aws/command_validate.go]: This was just leading to output like "Error creating cluster: Error: Key does not exist".  I think the "Validating stack template..." output added in the validate command makes the extra messaging unnecessary.

File: multi-node/aws/pkg/config/config.go (hunk 1 above)
Comment (cgag) [inline multi-node/aws/pkg/config/config.go]: done

File: multi-node/aws/cmd/kube-aws/command_validate.go (hunk 2 above)
Comment (cgag) [inline multi-node/aws/cmd/kube-aws/command_validate.go]: done

**Comment by cgag** [comment]
Moved the validateKey call to cluster.Create after an offline discussion with @colhom.

File: multi-node/aws/pkg/cluster/cluster.go [hunk 4]
```diff
@@ -254,3 +263,22 @@ func (c *Cluster) Destroy() error {
 	_, err := cfSvc.DeleteStack(dreq)
 	return err
 }
+
+func (c *Cluster) validateKey() error {
```
Comment (colhom) [inline multi-node/aws/pkg/cluster/cluster.go]: Check out how `validateExistingVPCState` is [set up](`validateExistingVPCState` takes an ec2

Comment (cgag) [inline multi-node/aws/pkg/cluster/cluster.go]: tested

File: multi-node/aws/pkg/cluster/cluster_test.go [hunk 5]
```diff
@@ -1,24 +1,40 @@
...
+type dummyEC2Service struct {
+	VPCs     map[string]VPC
+	KeyPairs map[string]bool
+}
+
 func (svc dummyEC2Service) DescribeVpcs(input *ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error) {
 	output := ec2.DescribeVpcsOutput{}
-	for _, vpcId := range input.VpcIds {
-		if vpc, ok := svc[*vpcId]; ok {
+	for _, vpcID := range input.VpcIds {
```
Comment (cgag) [inline multi-node/aws/pkg/cluster/cluster_test.go]: This capitalization makes golint happy, but it's inconsistent with the amazon's naming.  I'd prefer to not propagate their deviancy, but I can change this back if we'd rather be consistent.

Comment (colhom) [inline multi-node/aws/pkg/cluster/cluster_test.go]: golint is our source of truth in these matters.

**Comment by colhom** [comment]
LGTM

**Pull request merged**